### PR TITLE
AP_HAL_ChibiOS:added new hardware PixPilot-V6PRO

### PR DESCRIPTION
**Pull request by mikefenghao**
AP_HAL_ChibiOS:added new hardware PixPilot-V6PRO

### CODE REVIEW

**Comment by peterbarker** [comment]
Commit message for the bootloader isn't quite right; should have  `Tools: or `bootloaders:` on the front

**Comment by mikefenghao** [comment]
> Commit message for the bootloader isn't quite right; should have `Tools: or `bootloaders:` on the front

Thanks for pointing out the problem, I have revised it.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+
+•	internal Soft Rubber Damping Ball isolation for All interna IMUs
+
+•	Two barometers, BMP388(SPI)
+
+•	builtin RAMTRON(SPI)
+
+•	microSD card slot
+
+•	5 UARTs
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:21]:
```suggestion
•	5 UARTs, two with RTS/CTS flow control
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 2]
```diff
@@ -0,0 +1,352 @@
+## PixPilot-V6PRO Flight Controller
+
+The PixPilot-V6PRO flight controller is sold by a range of resellers listed on the makeflyeasy(http://www.makeflyeasy.com)
+
+## Features
+
+•	STM32H743VIT6 and STM32F103C8T6 microcontroller
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:7]:
```suggestion
•	STM32H743VIT6 microcontroller
*  STM32F103C8T6 IOMCU microcontroller
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+•	5 UARTs
+
+•	USB(Type-C)
+
+•	PPM & S.Bus input
+
+•	16 PWM outputs
+
+•	tow I2C ports and two FDCAN ports
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:29]:
```suggestion
•	Two I2C ports and two FDCAN ports
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+•	one S.Bus output
+
+•	internal Buzzer
+
+•	builtin RGB LED
+
+•	two voltage & current monitoring
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:37]:
```suggestion
•	Four voltage & current monitoring, Two analog and Two CAN
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+![PixPilot-V6PRO](PixPilot-V6Pro-1.png "PixPilot-V6Pro-1")
+![PixPilot-V6PRO](PixPilot-V6Pro-2.png "PixPilot-V6Pro-2")
+
+## Pinout
+
+UART Mapping
+============
+
+ - SERIAL0 -> console (primary mavlink, usually USB)
+ - SERIAL1 -> USART2  (telem1)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:55]:
```suggestion
 - SERIAL1 -> USART2  (Telem1,MAVLINK2) (DMA capable)
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+ - SERIAL2 -> USART3  (Telem2)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:56]:
```suggestion
 - SERIAL2 -> USART3  (Telem2, MAVLink2) (DMA capable)
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+ - SERIAL3 -> UART4   (GPS1)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:57]:
```suggestion
 - SERIAL3 -> UART4   (GPS1)  (TX is DMA capable)
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+ - SERIAL4 -> UART8   (GPS2)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:58]:
```suggestion
 - SERIAL4 -> UART8   (GPS2) (RX is DMA capable)
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+   <td>RX (IN)</td>
+   <td>+3.3V</td>
+   </tr>
+   <tr>
+   <td>4</td>
+   <td>GND</td>
+   <td>GND</td>
+   </tr>
+   </tbody>
+   </table>
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:266]:
```suggestion
   </table>
   
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+Power1, Power2 ports
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:267]: Missing the CAN power ports, and DSM port

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md [hunk 11]
```diff
@@ -0,0 +1,352 @@
...
+   <td>+5V</td>
+   </tr>
+   <tr>
+   <td>2</td>
+   <td>LED</td>
+   <td>+5V</td>
+   </tr>
+   <tr>
+   <td>3</td>
+   <td>SAFKEY</td>
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:185]:
```suggestion
   <td>Safety Switch</td>
```

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 10 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:267]: Also where are the two analog inputs PC3/4?

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat [hunk 12]
```diff
@@ -0,0 +1,285 @@
...
+PC1 42688_EXT_DRDY INPUT
+PC15 40605_EXT_DRDY INPUT
+PD15 42688_DRDY INPUT
+
+# Now we start defining some PWM pins. We also map these pins to GPIO
+# values, so users can set SERVOx_FUNCTION=-1 to determine which
+# PWM outputs on the primary MCU are set up as GPIOs.
+# To match HAL_PX4 we number the GPIOs for the PWM outputs
+# starting at 50.
+PE14 TIM1_CH4 TIM1 PWM(1) GPIO(50)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat:154]: I think that output 1-6 can be declared be setup as BIDIR...I tried locally and TIM1 and TIM4 have nonshared DMA so should be possible

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+define HAL_OS_FATFS_IO 1
+
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+
+
+# Now setup the default battery pins driver analog pins and default
+# scaling for the power brick.
+define HAL_BATT_MONITOR_DEFAULT 4
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat:235]: Are you going to setup Power2?

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef-bl.dat [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+PD7 BARO_EXT_CS CS
+PC2 40605_EXT_CS CS
+PE4 42688_CS CS
+PC14 BARO_CS CS 
+PD10 FRAM_CS CS
+#PD2 SDCARD_CS CS
+
+
+define HAL_USE_EMPTY_STORAGE 1
+define HAL_STORAGE_SIZE 32768
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef-bl.dat:60]: need blank line at file end

also should add SD card to allow firmware load from SD

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
I would also add a default.param file setting up the two CAN drivers...ie CAN_P1_DRIVER =1,CAN_P2_DRIVER=1 so that they appear active when inspected

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat (hunk 12 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat:154]: PE14 TIM1_CH4 TIM1 PWM(1) GPIO(50)
PE13 TIM1_CH3 TIM1 PWM(2) GPIO(51) BIDIR
PE11 TIM1_CH2 TIM1 PWM(3) GPIO(52)
PE9  TIM1_CH1 TIM1 PWM(4) GPIO(53) BIDIR
PD13 TIM4_CH2 TIM4 PWM(5) GPIO(54) BIDIR
PD14 TIM4_CH3 TIM4 PWM(6) GPIO(55) BIDIR

PB0 TIM3_CH3 TIM3 PWM(7) GPIO(56)  NODMA
PB1 TIM3_CH4 TIM3 PWM(8) GPIO(57)  NODMA

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat [hunk 15]
```diff
@@ -0,0 +1,285 @@
...
+PC0 VBUS_nVALID INPUT PULLUP
+
+
+# telem1
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+PD3 USART2_CTS USART2
+PD4 USART2_RTS USART2
+
+PD3 EXTERN_GPIO4 OUTPUT GPIO(4) ALT(1)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat:44]: We are not 100% certain this works....test thoroughly...ie use alt config and test relay outputs can be controlled using these gpios

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 1 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md]: I have modified and added default.param.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 2 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md]: I've changed it as you said.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 3 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md]: I have corrected the wrong spelling.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 4 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md]: Have been modified

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 5 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md]: Have been modified

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 6 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md]: Have been modified

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 7 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md]: Have been modified

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 8 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md]: Have been modified

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 9 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:339]: Have been modified

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md (hunk 10 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/README.md:341]: I have added  CAN power ports and DSM port。 the two analog inputs PC3/4 on the power2 ports.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat (hunk 12 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat:156]: Have been modified，Thanks.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat (hunk 13 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat:238]: I've added power2.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef-bl.dat (hunk 14 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef-bl.dat]: OK，Have been modified.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat (hunk 15 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat]: now.I removed the EXTERN_GPIO function of PD3,PD4.

**Review by Hwurzburg — APPROVED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef-bl.dat [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+FLASH_RESERVE_START_KB 0
+
+# the location where the bootloader will put the firmware
+# the H743 has 128k sectors
+FLASH_BOOTLOADER_LOAD_KB 128
+
+env OPTIMIZE -Os
+PB3 LED_ACTIVITY OUTPUT OPENDRAIN HIGH # red
+PE12 LED_BOOTLOADER OUTPUT OPENDRAIN HIGH # blue
+define HAL_LED_ON 0
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef-bl.dat:25]: remove... will now cause build error

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat [hunk 17]
```diff
@@ -0,0 +1,296 @@
...
+define HAL_I2C_INTERNAL_MASK 1
+
+undef AP_FEATURE_SBUS_OUT
+
+
+# Enable FAT filesystem support.
+define HAL_OS_FATFS_IO 1
+AP_BOOTLOADER_FLASH_FROM_SD_ENABLED 
+
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
```
[the remark this answers is not in the document]
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat]: I have deleted.

File: libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef-bl.dat (hunk 16 above)
Comment (mikefenghao) [reply libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef-bl.dat:25]: Yes, if I remove it, it will build error. Now it works fine.

**Review by Hwurzburg — APPROVED** [review]

**Pull request merged**